server/lib/systemd/pbench-server.service

```diff
@@ -14,16 +14,11 @@ User = pbench
 Group = pbench
 RuntimeDirectory = pbench-server
 PIDFile = /run/pbench-server/gunicorn.pid
-ExecStart = /opt/pbench-server/bin/pbench-server
+ExecStart = /bin/bash -c 'PATH=${PATH}:$(python3 -m site --user-base)/bin /opt/pbench-server/bin/pbench-server'
```

Comment:
Seems like this is a bit delicate and nuanced.  Systemd offers a method of setting `PATH`, perhaps doing something like https://stackoverflow.com/questions/35439123/systemd-environment-directive-to-set-path?

Reply:
@portante: that  can only deal with a constant value - there is no mechanism for substituting the previous value of PATH or of the value of the command. Right?

Reply:
Sure, but does the output of `python3 -m site --user-base` change each time it is invoked?  If not, why not record that once in an environment variable file at install time?

Reply:
The overhead of firing up `bash` once on server start didn't seem like a big deal vs going to the trouble of modifying the `pbench-server.service` file on installation... although I did consider that.

No, the output won't change each time, but it is tied to the creation of the `pbench` user account and the host/account's default `PATH`. E.g., I could have assumed something like `/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/home/pbench/.local/bin`, but decided I'd rather not do that.

Reply:
> Seems like this is a bit delicate and nuanced.

Nuanced, yes; delicate, not so much.  This approach seems adequate to me, and it needs to be evaluated in the context of the `pbench` user account, and I think this accomplishes that in an unequivocal way.  (I may steal this for the containerized testing stuff that I'm working on.)

Reply:
As mentioned here, https://serverfault.com/questions/413397/how-to-set-environment-variable-in-systemd-service, for Fedora, the recommended why to write systemd service files is to use `EnvironmentFile`.  On our path to productization is a stop along the way where we get our code to land in Fedora.

Reply:
Separate from that, what is the requirement that `PATH` needs to be updated for `/opt/pbench-server/bin/pbench-server` to work?

Reply:
Any can't find anywhere in the code on the server side where `PATH` needs to be updated.

Reply:
Also filed https://github.com/dbutenhof/pbench/pull/2.

Reply:
> Any can't find anywhere in the code on the server side where `PATH` needs to be updated.

It's right in the commit message: "Fix the pbench-server service file to set up a proper PATH to find a pbench --user installed gunicorn binary".

When `gunicorn` is installed via our `requirements.txt` now, it goes into `/home/pbench/.local/bin` and, because that's not on the default `PATH`, the `pbench-server` service fails to start. (Yes, the wrapper ends up actually executing the `gunicorn` binary in a subprocess, which needs the `PATH`.)

There are multiple ways around this; this seemed the most consistent with our current approach. In the longer term, especially with Fedora migration, we know we need to dump `requirements.txt` and move to RPM dependencies, which would avoid the whole `PYTHONPATH` issue as well as `PATH`. As always, the question is how much we want to do how quickly.

Reply:
Why not set it in `./lib/pbench/cli/server/shell.py`?